api/portainer.go

```diff
@@ -664,6 +664,20 @@ type (
 	// RoleID represents a role identifier
 	RoleID int
 
+	// APIKeyID represents an API key identifier
+	APIKeyID int
+
+	// APIKey represents an API key
+	APIKey struct {
+		ID          APIKeyID `json:"Id" example:"1"`
+		UserID      UserID   `json:"UserId" example:"1"`
+		Description string
+		Prefix      string    `json:"prefix"`           // API key identifier (3 char prefix)
+		DateCreated time.Time `json:"dateCreated"`      // Date when the API key was created (UTC)
+		LastUsed    time.Time `json:"lastUsed"`         // Date when the API key was last used (UTC)
+		Digest      []byte    `json:"digest,omitempty"` // Digest represents the hash of the raw API key
```

Comment:
`raw API key` is the un-hashed API key - before we hash it.
Feel free to suggest another name.

The intention was that the hashing algorithm are implementation details - this struct should not be linked to a specific hash algorithm.
^ the service interfaces return this struct - hence why i think it should not specifically define the algorithm.

note: this can be changed though (we can discuss)